src/subscription/RelaySubscription.js

```diff
@@ -0,0 +1,387 @@
+   *    response that contains the newly created edge, `rangeBehaviors` maps
+   *    stringified representation of calls on the connection to
+   *    GraphQLMutatorConstants.RANGE_OPERATIONS.
+   *
+   * -  NODE_DELETE provides configuration for deleting a node and the
+   *    corresponding edge from a range.
+   *    {
+   *      type: RelayMutationType.NODE_DELETE;
+   *      parentName: string;
+   *      parentID: string;
+   *      connectionName: string;
+   *      deletedIDFieldName: string;
+   *    }
+   *    where `parentName`, `parentID` and `connectionName` refer to the same
+   *    things as in RANGE_ADD, `deletedIDFieldName` is the name of the key in
+   *    the server response that contains the DataID of the deleted node.
+   *
+   * -  RANGE_DELETE provides configuration for deleting an edge from a range
+   *    but doesn't delete the node.
+   *    {
+   *      type: RelayMutationType.RANGE_DELETE;
+   *      parentName: string;
+   *      parentID: string;
+   *      connectionName: string;
+   *      deletedIDFieldName: string | Array<string>;
+   *      pathToConnection: Array<string>;
+   *    }
+   *    where `parentName`, `parentID`, `connectionName` and
+   *    `deletedIDFieldName` refer to the same things as in NODE_DELETE.
+   *    `deletedIDFieldName` can also be a path from the response root to the
+   *    deleted node. `pathToConnection` is a path from `parentName` to
+   *    `connectionName`.
+   *
+   * -  REQUIRED_CHILDREN is used to append additional children (fragments or
+   *    fields) to the subscription query. Any data fetched for these children is
+   *    not written to the client store, but you can add code to process it
+   *    in the `onSuccess` callback passed to the `RelayEnvironment` `applyUpdate`
+   *    method. You may need to use this, for example, to fetch fields on a new
+   *    object created by the subscription (and which Relay would normally not
+   *    attempt to fetch because it has not previously fetched anything for that
+   *    object).
+   *    {
+   *      type: RelayMutationType.REQUIRED_CHILDREN;
+   *      children: Array<RelayQuery.Node>;
+   *    }
+   */
+  getConfigs(): Array<RelayMutationConfig> {
+    invariant(
+      false,
+      '%s: Expected abstract method `getConfigs` to be implemented.',
+      this.constructor.name
+    );
+  }
+
+  /**
+   * These variables form the "input" to the subscription query sent to the server.
+   */
+  getVariables(): {[name: string]: mixed} {
+    invariant(
+      false,
+      '%s: Expected abstract method `getVariables` to be implemented.',
+      this.constructor.name
+    );
+  }
+
+  /**
+   * An optional collision key allows a subscription to identify itself with other
+   * subscriptions that affect the same fields. Subscriptions with the same collision
+   * are sent to the server serially and in-order to avoid unpredictable and
+   * potentially incorrect behavior.
+   */
+  getCollisionKey(): ?string {
```

Comment:
Ah, I see. Would probably be nice to have something that can de-duplicate subscriptions that listens on the same data. 👍